galaxy/api/access.py

```diff
@@ -391,6 +391,30 @@ def can_delete(self, obj):
         return False
 
 
+class EmailConfirmationAccess(BaseAccess):
+    # Emails are verified via GET request. This means that emails can be
+    # verified without requiring authentication if a confirmation key exists.
+    def can_read(self, obj):
+        return True
+
+    def can_add(self, data):
+        if self.user.is_authenticated():
+            if data.get('email_address') is None:
+                return True
+            email = EmailAddress.objects.get(pk=data.get('email_address'))
```

Comment:
This should be wrapped in a *try/except* that catches *ObjectDoesNotExist*. Otherwise, it's going to fail in with 500 error, I think.